client/scss/components/_footer.scss

```diff
@@ -85,12 +85,31 @@
   width: 100%;
 
   @include media-breakpoint-up(sm) {
-    width: 250px;
+    width: theme('spacing.64');
   }
 
   &--primary {
+    // Support basic layout support for items added via `extra_footer_actions`
+
+    align-items: stretch;
+    display: grid;
+    gap: theme('spacing.2');
+    grid-auto-flow: column;
+
+    > * {
+      // Reset the margin on any .button sibling elements
+      height: auto;
```

Comment:
Unfortunately this causes the button to be very small when there's no dropdown (e.g. when the page is locked)

<img width="670" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/070697df-f197-4c74-bd0b-cd5019983b17">

and the button to be very large when it has an icon (due to missing `height` set for the icon):

<img width="596" alt="image" src="https://github.com/wagtail/wagtail/assets/6379424/da488a3d-7522-4164-861b-96a2a52ed933">

I think we have to define our own height here instead of relying on the dropdown's button height, which in turn relies on the default `.button` height of `3em` or the `$text-input-height`:

https://github.com/wagtail/wagtail/blob/0599a56d817cc0a728b85ec936ea1c1d62cf19c9/client/scss/components/_dropdown-button.scss#L66

Unfortunately, 42px or 2.625rem (the current height, which is also `$text-input-height`) is not defined in our spacing tokens. We can either add it as `spacing[10.5]` and use it, or use `$text-input-height` instead.

Since we already use `$text-input-height` in `_dropdown-button.scss`, I think we can use it here too. It would be nice to add it as `spacing[10.5]` in the future though, as it seems we use this value in quite a few places.